### PR TITLE
[6.5] UI Port compute resource RHV tests

### PR DESCRIPTION
**Pull request by ldjebran**
Partially close https://github.com/SatelliteQE/robottelo/issues/6367 
Dependency:  https://github.com/SatelliteQE/airgun/pull/254
```
(robottelo) dlezz@dlezz:~/projects/robottelo-fork$ pytest -v tests/foreman/ui_airgun/test_computeresource.py tests/foreman/ui_airgun/test_computeresource_ec2.py
================================================================ test session starts ================================================================
platform linux -- Python 3.6.6, pytest-4.0.1, py-1.5.4, pluggy-0.8.0 -- /home/dlezz/.pyenv/versions/robottelo/bin/python3.6
cachedir: .pytest_cache
shared_function enabled - ON - scope: ffff - storage: redis
rootdir: /home/dlezz/projects/robottelo-fork, inifile:
plugins: wait-for-1.0.9, xdist-1.22.5, services-1.3.0, mock-1.10.0, forked-0.2, cov-2.5.1
collecting ... 2018-12-07 19:54:06 - conftest - DEBUG - BZ deselect is disabled in settings

collected 18 items                                                                                                                                  

tests/foreman/ui_airgun/test_computeresource.py::test_positive_end_to_end[True] PASSED                                                        [  5%]
tests/foreman/ui_airgun/test_computeresource.py::test_positive_end_to_end[False] PASSED                                                       [ 11%]
tests/foreman/ui_airgun/test_computeresource.py::test_positive_add_resource[True] PASSED                                                      [ 16%]
tests/foreman/ui_airgun/test_computeresource.py::test_positive_add_resource[False] PASSED                                                     [ 22%]
tests/foreman/ui_airgun/test_computeresource.py::test_positive_edit_resource_description[True] PASSED                                         [ 27%]
tests/foreman/ui_airgun/test_computeresource.py::test_positive_edit_resource_description[False] PASSED                                        [ 33%]
tests/foreman/ui_airgun/test_computeresource.py::test_positive_list_resource_vms[True] PASSED                                                 [ 38%]
tests/foreman/ui_airgun/test_computeresource.py::test_positive_list_resource_vms[False] PASSED                                                [ 44%]
tests/foreman/ui_airgun/test_computeresource.py::test_positive_edit_resource_version PASSED                                                   [ 50%]
tests/foreman/ui_airgun/test_computeresource.py::test_positive_resource_vm_power_management[True] PASSED                                      [ 55%]
tests/foreman/ui_airgun/test_computeresource.py::test_positive_resource_vm_power_management[False] PASSED                                     [ 61%]
tests/foreman/ui_airgun/test_computeresource.py::test_positive_update_organization[True] PASSED                                               [ 66%]
tests/foreman/ui_airgun/test_computeresource.py::test_positive_update_organization[False] PASSED                                              [ 72%]
tests/foreman/ui_airgun/test_computeresource.py::test_positive_image_end_to_end PASSED                                                        [ 77%]
tests/foreman/ui_airgun/test_computeresource.py::test_positive_associate_with_custom_profile PASSED                                           [ 83%]
tests/foreman/ui_airgun/test_computeresource.py::test_positive_associate_with_custom_profile_with_template PASSED                             [ 88%]
tests/foreman/ui_airgun/test_computeresource_ec2.py::test_positive_default_end_to_end_with_custom_profile PASSED                              [ 94%]
tests/foreman/ui_airgun/test_computeresource_ec2.py::test_positive_create_ec2_with_custom_region PASSED                                       [100%]

============================================================= 18 passed in 1582.72 seconds ============================================================
```

### CODE REVIEW

**Comment by ldjebran** [comment]
Test output to latest changes
```
(robottelo) dlezz@dlezz:~/projects/robottelo-fork$ pytest -v tests/foreman/ui_airgun/test_computeresource.py tests/foreman/ui_airgun/test_computeresource_ec2.py
============================================ test session starts =============================================
platform linux -- Python 3.6.6, pytest-4.0.1, py-1.5.4, pluggy-0.8.0 -- /home/dlezz/.pyenv/versions/robottelo/bin/python3.6
cachedir: .pytest_cache
shared_function enabled - ON - scope: ffff - storage: redis
rootdir: /home/dlezz/projects/robottelo-fork, inifile:
plugins: wait-for-1.0.9, xdist-1.22.5, services-1.3.1, mock-1.10.0, forked-0.2, cov-2.5.1
collecting ... 2018-12-11 12:28:49 - conftest - DEBUG - BZ deselect is disabled in settings

collected 18 items                                                                                           

tests/foreman/ui_airgun/test_computeresource.py::test_positive_end_to_end[True] PASSED                 [  5%]
tests/foreman/ui_airgun/test_computeresource.py::test_positive_end_to_end[False] PASSED                [ 11%]
tests/foreman/ui_airgun/test_computeresource.py::test_positive_add_resource[True] PASSED               [ 16%]
tests/foreman/ui_airgun/test_computeresource.py::test_positive_add_resource[False] PASSED              [ 22%]
tests/foreman/ui_airgun/test_computeresource.py::test_positive_edit_resource_description[True] PASSED  [ 27%]
tests/foreman/ui_airgun/test_computeresource.py::test_positive_edit_resource_description[False] PASSED [ 33%]
tests/foreman/ui_airgun/test_computeresource.py::test_positive_list_resource_vms[True] PASSED          [ 38%]
tests/foreman/ui_airgun/test_computeresource.py::test_positive_list_resource_vms[False] PASSED         [ 44%]
tests/foreman/ui_airgun/test_computeresource.py::test_positive_edit_resource_version PASSED            [ 50%]
tests/foreman/ui_airgun/test_computeresource.py::test_positive_resource_vm_power_management[True] PASSED [ 55%]
tests/foreman/ui_airgun/test_computeresource.py::test_positive_resource_vm_power_management[False] PASSED [ 61%]
tests/foreman/ui_airgun/test_computeresource.py::test_positive_update_organization[True] PASSED        [ 66%]
tests/foreman/ui_airgun/test_computeresource.py::test_positive_update_organization[False] PASSED       [ 72%]
tests/foreman/ui_airgun/test_computeresource.py::test_positive_image_end_to_end PASSED                 [ 77%]
tests/foreman/ui_airgun/test_computeresource.py::test_positive_associate_with_custom_profile PASSED    [ 83%]
tests/foreman/ui_airgun/test_computeresource.py::test_positive_associate_with_custom_profile_with_template PASSED [ 88%]
tests/foreman/ui_airgun/test_computeresource_ec2.py::test_positive_default_end_to_end_with_custom_profile PASSED [ 94%]
tests/foreman/ui_airgun/test_computeresource_ec2.py::test_positive_create_ec2_with_custom_region PASSED [100%]
================================= 18 passed in 1869.26 seconds ==================================
(robottelo) dlezz@dlezz:~/projects/robottelo-fork$


```

**Comment by ldjebran** [comment]
```
(robottelo) dlezz@dlezz:~/projects/robottelo-fork$ pytest -v tests/foreman/ui_airgun/test_computeresource.py tests/foreman/ui_airgun/test_computeresource_ec2.py
============================================ test session starts =============================================
platform linux -- Python 3.6.6, pytest-4.0.1, py-1.5.4, pluggy-0.8.0 -- /home/dlezz/.pyenv/versions/3.6.6/envs/robottelo/bin/python
cachedir: .pytest_cache
shared_function enabled - ON - scope: ffff - storage: redis
rootdir: /home/dlezz/projects/robottelo-fork, inifile:
plugins: wait-for-1.0.9, xdist-1.22.5, services-1.3.1, mock-1.10.0, forked-0.2, cov-2.5.1
collecting ... 2018-12-13 13:49:26 - conftest - DEBUG - BZ deselect is disabled in settings

collected 18 items                                                                                           

tests/foreman/ui_airgun/test_computeresource.py::test_positive_end_to_end[True] PASSED                 [  5%]
tests/foreman/ui_airgun/test_computeresource.py::test_positive_end_to_end[False] PASSED                [ 11%]
tests/foreman/ui_airgun/test_computeresource.py::test_positive_add_resource[True] PASSED               [ 16%]
tests/foreman/ui_airgun/test_computeresource.py::test_positive_add_resource[False] PASSED              [ 22%]
tests/foreman/ui_airgun/test_computeresource.py::test_positive_edit_resource_description[True] PASSED  [ 27%]
tests/foreman/ui_airgun/test_computeresource.py::test_positive_edit_resource_description[False] PASSED [ 33%]
tests/foreman/ui_airgun/test_computeresource.py::test_positive_list_resource_vms[True] PASSED          [ 38%]
tests/foreman/ui_airgun/test_computeresource.py::test_positive_list_resource_vms[False] PASSED         [ 44%]
tests/foreman/ui_airgun/test_computeresource.py::test_positive_edit_resource_version PASSED            [ 50%]
tests/foreman/ui_airgun/test_computeresource.py::test_positive_resource_vm_power_management[True] PASSED [ 55%]
tests/foreman/ui_airgun/test_computeresource.py::test_positive_resource_vm_power_management[False] PASSED [ 61%]
tests/foreman/ui_airgun/test_computeresource.py::test_positive_update_organization[True] PASSED        [ 66%]
tests/foreman/ui_airgun/test_computeresource.py::test_positive_update_organization[False] PASSED       [ 72%]
tests/foreman/ui_airgun/test_computeresource.py::test_positive_image_end_to_end PASSED                 [ 77%]
tests/foreman/ui_airgun/test_computeresource.py::test_positive_associate_with_custom_profile PASSED    [ 83%]
tests/foreman/ui_airgun/test_computeresource.py::test_positive_associate_with_custom_profile_with_template PASSED [ 88%]
tests/foreman/ui_airgun/test_computeresource_ec2.py::test_positive_default_end_to_end_with_custom_profile PASSED [ 94%]
tests/foreman/ui_airgun/test_computeresource_ec2.py::test_positive_create_ec2_with_custom_region PASSED [100%]

============================================== warnings summary ==============================================
tests/foreman/ui_airgun/test_computeresource.py::test_positive_end_to_end[True]
  /home/dlezz/.pyenv/versions/3.6.6/envs/robottelo/lib/python3.6/site-packages/selenium/webdriver/chrome/webdriver.py:50: DeprecationWarning: use options instead of chrome_options
    warnings.warn('use options instead of chrome_options', DeprecationWarning)
  /home/dlezz/.pyenv/versions/3.6.6/envs/robottelo/lib/python3.6/site-packages/selenium/webdriver/remote/webdriver.py:796: DeprecationWarning: use driver.switch_to.alert instead
    warnings.warn("use driver.switch_to.alert instead", DeprecationWarning)
  /home/dlezz/.pyenv/versions/3.6.6/envs/robottelo/lib/python3.6/site-packages/selenium/webdriver/remote/webdriver.py:796: DeprecationWarning: use driver.switch_to.alert instead
    warnings.warn("use driver.switch_to.alert instead", DeprecationWarning)
...

-- Docs: https://docs.pytest.org/en/latest/warnings.html
================================= 18 passed, 42 warnings in 1388.22 seconds ==================================
```

File: tests/foreman/ui_airgun/test_computeresource.py [hunk 1]
```diff
@@ -206,3 +268,256 @@ def test_positive_resource_vm_power_management(
...
+            {'provider_content.{0}'.format(key): value for key, value in cr_profile_data.items()}
+        )
+        provider_content_values = session.computeresource.read_computeprofile(
+            cr_name, COMPUTE_PROFILE_LARGE)['provider_content']
+        # assert main compute resource profile data updated updated successfully.
+        excluded_keys = ['network_interfaces', 'storage']
+        assert (
+            {key: value for key, value
+             in cr_profile_data.items() if key not in excluded_keys}
+            ==
```
Comment (abalakh) [inline tests/foreman/ui_airgun/test_computeresource.py]: non-blocking suggestion: this assertion and the next one are pretty hard to read. Would you mind storing these dicts in some `expected_data`/`actual_data` variables (or smth similar)?

**Comment by ldjebran** [comment]
test output for last changes
```
(robottelo) dlezz@dlezz:~/projects/robottelo-fork$ pytest -v tests/foreman/ui_airgun/test_computeresource.py::test_positive_associate_with_custom_profile
============================================ test session starts =============================================
platform linux -- Python 3.6.6, pytest-3.6.1, py-1.5.4, pluggy-0.6.0 -- /home/dlezz/.pyenv/versions/robottelo/bin/python3.6
cachedir: .pytest_cache
shared_function enabled - ON - scope: ffff - storage: redis
rootdir: /home/dlezz/projects/robottelo-fork, inifile:
plugins: wait-for-1.0.9, xdist-1.22.5, services-1.3.1, mock-1.10.0, forked-0.2, cov-2.5.1
collecting 1 item                                                                                            2018-12-18 18:17:16 - conftest - DEBUG - BZ deselect is disabled in settings

collected 1 item                                                                                             

tests/foreman/ui_airgun/test_computeresource.py::test_positive_associate_with_custom_profile PASSED    [100%]

========================================= 1 passed in 130.21 seconds =========================================
```

File: tests/foreman/ui_airgun/test_computeresource.py (hunk 1 above)
Comment (ldjebran) [reply tests/foreman/ui_airgun/test_computeresource.py]: implemented

**Comment by ldjebran** [comment]
@san7ket @abalakh comment addressed and rebased

**Pull request merged**